cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -955,6 +950,10 @@ func (s *Supervisor) onMessage(ctx context.Context, msg *types.MessageData) {
 	}
 }
 
+func (s *Supervisor) persistentStateFile() string {
+	return filepath.Join(s.config.Storage.DirectoryOrDefault(), persistentStateFile)
```

Comment:
Should we `mkdir` the directory in case it does not exist?

Reply:
Yeah, we should. I've added the directory creation logic in the `NewSupervisor` function.